### PR TITLE
expression: refine built-in func truncate to support uint arg

### PR DESCRIPTION
**Pull request by yu34po**
### What problem does this PR solve? 
fix https://github.com/pingcap/tidb/issues/7591
fix https://github.com/pingcap/tidb/issues/7615

### What is changed and how it works?
1.add an builtinTruncateUintSig Function to support unsigned int truncate 
2.change builtinTruncateIntSig truncate function to avoid get incorrect value
### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects
None

Related changes

 - Need to cherry-pick to the release branch
release-2.0 release-2.1

### CODE REVIEW

File: expression/builtin_math.go [hunk 1]
```diff
@@ -1826,6 +1831,41 @@ func (b *builtinTruncateIntSig) evalInt(row chunk.Row) (int64, bool, error) {
...
+	}
+}
+
+func (b *builtinTruncateUnintSig) Clone() builtinFunc {
+	newSig := &builtinTruncateUnintSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+type builtinTruncateUnintSig struct {
```
Comment (zz-jason) [inline expression/builtin_math.go]: s/Unint/Uint/

**Comment by zz-jason** [comment]
@yu34po please check the `Check List` carefully, remove the unnecessary items.

File: expression/builtin_math_test.go [hunk 2]
```diff
@@ -485,6 +485,9 @@ func (s *testEvaluatorSuite) TestTruncate(c *C) {
 		{[]interface{}{newDec("23.298"), -100}, newDec("0")},
 		{[]interface{}{newDec("23.298"), 100}, newDec("23.298")},
 		{[]interface{}{nil, 2}, nil},
+		{[]interface{}{uint64(9223372036854775808), -10}, 9223372030000000000},
```
Comment (XuHuaiyu) [inline expression/builtin_math_test.go]: We also need to add some test cases in 
expression/integration_test.go: line476~479

**Comment by zz-jason** [comment]
@yu34po would you please help us to cherrypick this commit to the **release-2.1** and **release-2.0** branches?

**Comment by yu34po** [comment]
@zz-jason cannot do it  as my local commit has been deleted..

**Comment by zz-jason** [comment]
@yu34po You can find the commit in the tidb master branch and use `git cherry-pick $commit_hash` to cherry pick that commit to the release-2.1 branch.

**Pull request merged**